presto-native-execution/presto_cpp/main/operators/FileBroadcast.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/operators/FileBroadcast.h"
+#include <boost/lexical_cast.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include "velox/serializers/PrestoSerializer.h"
+using namespace facebook::velox::exec;
+using namespace facebook::velox;
+
+namespace facebook::presto::operators {
+
+FileBroadcast::FileBroadcast(const std::string& basePath)
+    : basePath_(basePath) {
+  VELOX_CHECK(!basePath.empty(), "Root path for broadcast files is empty!");
+  fileSystem_ = velox::filesystems::getFileSystem(basePath, nullptr);
+}
+
+std::shared_ptr<BroadcastFileWriter> FileBroadcast::createWriter(
+    memory::MemoryPool* pool,
+    const RowTypePtr& inputType) {
+  fileSystem_->mkdir(basePath_);
+  auto filename =
+      fmt::format("{}/file_broadcast_{}.bin", basePath_, makeUuid());
+  LOG(INFO) << "Opening broadcast file for write: " << filename;
+  auto writeFile = fileSystem_->openFileForWrite(filename);
+  auto broadcastFileWriter = std::make_shared<BroadcastFileWriter>(
+      std::move(writeFile), filename, pool, inputType);
+  return broadcastFileWriter;
+}
+
+// static
+std::string FileBroadcast::makeUuid() {
+  return boost::lexical_cast<std::string>(boost::uuids::random_generator()());
+}
+
+BroadcastFileWriter::BroadcastFileWriter(
+    std::unique_ptr<WriteFile> writeFile,
+    std::string_view filename,
+    velox::memory::MemoryPool* pool,
+    const RowTypePtr& inputType)
+    : writeFile_(std::move(writeFile)),
+      filename_(filename),
+      pool_(pool),
+      serde_(std::make_unique<serializer::presto::PrestoVectorSerde>()),
+      inputType_(inputType) {}
+
+void BroadcastFileWriter::collect(RowVectorPtr input) {
+  serialize(input);
+}
+void BroadcastFileWriter::noMoreData() {
+  writeFile_->flush();
```

Comment:
I see that collect flushes on each call. Do we also need this additional flush?

Reply:
Its not needed